### ISSUE TITLE
serverIdleTimeOut does not work

### ISSUE THREAD

**Issue by lervag** [sbt/sbt#7631]
## problem

I've noticed that when I start `sbt` as a build server, it will remain on my system for a long time. I've asked and searched for information about this, and I ended up finding a few related issues:

* https://github.com/sbt/sbt/issues/6176
* https://github.com/sbt/sbt/issues/6536

From this, it seemed I could reduce the timeout delay by adding `serverIdleTimeOut` to my `global.sbt`. However, when I try that, I get the following error:

```
/home/lervag/.sbt/1.0/global.sbt:5: error: not found: value serverIdleTimeOut
Global / serverIdleTimeOut := Some(new FiniteDuration(5, TimeUnit.MINUTES))
```

I'll admit, it may be possible that I just misunderstand something essential. However, I don't find this option documented anywhere and the issue threads were ~3 years old or older. Thus, it is equally likely to me that this is a bug.

If I _am_ doing something wrong, then I would be very happy to get some assistance. And if I'm pointed to the right direction I might consider helping with suggesting a minor update of some relevant docs.

## steps

I have `~/.sbt/1.0/global.sbt` with this content:

```scala
import scala.concurrent.duration.FiniteDuration
import java.util.concurrent.TimeUnit

Global / serverIdleTimeOut := Some(new FiniteDuration(5, TimeUnit.MINUTES))
```

I have `sbt` version 1.10.1 installed. I do this:

```sh
cd $HOME
mkdir tmp
cd tmp
sbt -bsp
```

## expectation

I expect the build server to start properly, but it fails with the error message displayed above.

---

Let me know if more details are needed here.

**Comment by adpi2** [comment]
Could it be a typo? It should be serverIdleTime**o**ut instead of serverIdleTime**O**ut, as per: https://github.com/sbt/sbt/blob/2f936295f158117baf9397ce8e9487bf9be8c87e/main/src/main/scala/sbt/Keys.scala#L107

**Comment by lervag** [comment]
Omg, you're right. Although, it's not so strange that I was confused! Notice:

```
sbt:tmp> show Global/serverIdleTimeout
[error] Not a valid key: serverIdleTimeout (similar: serverIdleTimeOut, serverPort, serverHost)
sbt:tmp> show Global/serverIdleTimeOut
[info] Some(15 seconds)
```

It seems there is an inconsistency in the casing between _setting_ the option and showing its value. I'm not sure if that's worth fixing, but at least it makes it slightly less embarrassing.

Also, this is not documented anywhere, right? If you point me to the right place, I could propose some text for documentation.

**Comment by lervag** [comment]
It seems to me that the only inconsistency is in line 96 here:

https://github.com/sbt/sbt/blob/2f936295f158117baf9397ce8e9487bf9be8c87e/main-command/src/main/scala/sbt/BasicKeys.scala#L94-L96

When I search the project, the `TimeOut` version is only used a single time, whereas `Timeout` is used all the other places.

**Comment by adpi2** [comment]
> It seems there is an inconsistency in the casing between setting the option and showing its value. I'm not sure if that's worth fixing, but at least it makes it slightly less embarrassing.

Thanks for figuring this out. Yes it is probably worth fixing.

> Also, this is not documented anywhere, right? If you point me to the right place, I could propose some text for documentation.

No it is not documented anywhere. The documentation of the sbt server is [here](https://www.scala-sbt.org/1.x/docs/sbt-server.html#sbt+Server), but it is largely outdated. Feel free to add a mention of `serverIdleTimeout` though, it would still be better than nothing.

**Comment by lervag** [comment]
I've attempted to fix the inconsistency now (see #7651), and I've made a draft at the docs as well.

**Comment by Friendseeker** [comment]
Fixed by https://github.com/sbt/sbt/pull/7651
